src/Mvc/Mvc.TagHelpers/src/ComponentOutputTagHelper.cs

```diff
@@ -0,0 +1,56 @@
+// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using Microsoft.AspNetCore.Mvc.Rendering;
+using Microsoft.AspNetCore.Mvc.ViewFeatures;
+using Microsoft.AspNetCore.Razor.TagHelpers;
+
+namespace Microsoft.AspNetCore.Mvc.TagHelpers
+{
+    /// <summary>
+    /// A <see cref="TagHelper"/> used for emitting deferred component output.
+    /// </summary>
+    [HtmlTargetElement(TagHelperName, TagStructure = TagStructure.WithoutEndTag)]
+    public class ComponentOutputTagHelper : TagHelper
```

Comment:
Is there a way to make a bespoke HeadOutputTagHelper? The user doesn't get an opportunity to configure the section name elsewhere, so it'd be weird to spring this type on them.

Reply:
The user can optionally configure the section name in `<component>` tag helpers, e.g.:

```razor
<component output-name="app" type="typeof(HeadModification)" render-mode="ServerPrerendered" />
<component output-name="head" type="typeof(HeadRoot)" render-mode="ServerPrerendered" />

<!DOCTYPE html>
<html>
<head>
    <base href="~/" />
    <component-output name="head" />
</head>
<body>
    <app><component-output name="app" /></app>
</body>
</html>
```

Reply:
Ahh, I see how this works. It's not really tied to the section name.